tests/test_events.py

```diff
@@ -1,7 +1,50 @@
+import uuid
 from fastapi.testclient import TestClient
+from typing import Dict, Callable
 
 
-def test_post_events(client: TestClient):
-    response = client.post("/v1/events", json={})
+def test_post_events(
+    client_factory: Callable[[Dict], TestClient],
+    admin_header_factory: Callable[[str], Dict]
+):
+    auth_keys = [{"kid": "kid1", "secret": "secret1"},
+                 {"kid": "kid2", "secret": "secret2"}]
+    client = client_factory(auth_keys)
+    auth_header = admin_header_factory(str(uuid.uuid4()), "kid1", "secret1")
+    response = client.post("/v1/events", headers=auth_header)
     assert response.status_code == 201
     assert "detail" in response.json()
+
+
+def test_post_two_different_kids(
+    client_factory: Callable[[Dict], TestClient],
+    admin_header_factory: Callable[[str], Dict]
+):
+    auth_keys = [{"kid": "kid1", "secret": "secret1"},
+                 {"kid": "kid2", "secret": "secret2"}]
+    client = client_factory(auth_keys)
+
+    auth_header = admin_header_factory(str(uuid.uuid4()), "kid1", "secret1")
+    response = client.post("/v1/events", headers=auth_header)
+    assert response.status_code == 201
+    assert "detail" in response.json()
+
+    auth_header = admin_header_factory(str(uuid.uuid4()), "kid2", "secret2")
+    response = client.post("/v1/events", headers=auth_header)
+    assert response.status_code == 201
+    assert "detail" in response.json()
+
+
+def test_expired_jwt(
+    client_factory: Callable[[Dict], TestClient],
+    admin_header_factory: Callable[[str], Dict]
+):
+    auth_keys = [{"kid": "kid1", "secret": "secret1"},
+                 {"kid": "kid2", "secret": "secret2"}]
+    client = client_factory(auth_keys)
+
+    auth_header = admin_header_factory(
+        str(uuid.uuid4()), "key1", "secret1", expired=True
```

Comment:
I realized this after my earlier pass, but this test case is actually broken because it's returning a `403` for the wrong reason :joy: . You should fix this key id to be `kid1`. You may also want to add the following assertion to be sure you have the right error:

```python
assert response.json()['detail'] == 'Signature has expired.'
```

Note that when you fix this, you'll want to create a new test case for the `Invalid kid value in token` error path`